docs/DeveloperGuide.md

```diff
@@ -1,34 +1,503 @@
+
+### Print tasks feature
+This feature is facilitated by `PrintTasksCommand`.
+The following sequence diagram shows how the `execute()` operation works when the user wants to print the list of tasks stored in the program.
+![print_tasks_command_sd](images/PrintTasksCommand_SD.png)
+
+### Print events feature
+This feature is facilitated by `PrintEventsCommand`.
+The following sequence diagram shows how the `execute()` operation works when the user wants to print the list of events stored in the program.
+![print_events_command_sd](images/PrintEventsCommand_SD.png)
+
+### Print personal calendar feature
+This feature is facilitated by `PrintTimelineCommand`.
+The following sequence diagram show how the `execute()` operation works when the user wants to print the personalised calendar.:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/PrintTimelineCommand_SD.png" alt="" width="750"/><br>
+
+### Prioritize a task feature 
+This feature is facilitated by `PrioritizeCommand`. 
+The following sequence diagram shows how the `execute()` operation works when the user wants to prioritize a task:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/PrioritizeCommand_SD.png" alt="" width="750"/><br/>
+
+Note: It first extracts the task number from the user input prior to convertTaskNumberToCalendarNumber. 
+There will be a check in the function `markTaskAsImportant(calendarNumber)` to ensure that the calendar item being marked as important is a task.
+
+### Print prioritized tasks feature
+This feature is facilitated by `PrintPriorityCommand`.
+The following sequence diagram shows how the `execute()` operation works when the user wants to print all the prioritized tasks:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/PrintPriorityCommand_SD.png" alt="" width="750"/><br/>
+
+Note: It uses the `getIsImportant()` function of task items to identify whether it is a prioritized task.
+
+### Print progress feature
+This feature is facilitated by `PrintProgressCommand`.
+The following sequence diagram shows how the `execute()` operation works when the user wants to see the progress of all tasks:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/PrintProgressCommand_SD.png" alt="" width="500"/><br/>
+
+Note: It uses `calculateNumTotal()` and `calculateNumFinshed()` to calculate the numbers of only tasks and finished tasks in the list, not including events. 
+Then it uses these numbers to calculate the progress. 
+
+### Print suggestions feature
+This feature is facilitated by `PrintSuggestionCommand`.
+The following sequence diagram shows how the `execute()` operation works when the user wants to see suggestions about preparing which tasks:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/PrintSuggestionCommand_SD.png" alt="" width="500"/><br/>
```

Comment:
![image](https://user-images.githubusercontent.com/18654251/97660308-19886b80-1aac-11eb-9c95-d02bb1a22bee.png)
This diagram is a really good effort, however, it might be too complicated, as there is a lot of things going on. Perhaps, it would be beneficial to split it up into smaller parts, or focus on an individual aspect!